.github/workflows/optimize-ci.yml

```diff
@@ -297,17 +296,79 @@ jobs:
         with:
           archive_name: integration-tests-opensearch-dev-${{ matrix.includedGroups }}-docker
 
+  migration:
+    name: Migration
+    runs-on: gcp-core-8-default
+    timeout-minutes: 30
+    steps:
+      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4
+
+      - name: Setup Maven
+        uses: ./.github/actions/setup-maven
+        with:
+          secrets: ${{ toJSON(secrets) }}
+
+      - name: Login to Harbor registry
+        uses: ./.github/actions/login-registry
+        with:
+          secrets: ${{ toJSON(secrets) }}
+
+      - name: "Read Java / Version Info"
+        id: "pom-info"
+        uses: YunaBraska/java-info-action@main
+
+      - name: Start Elastic
+        uses: ./.github/actions/compose
+        with:
+          compose_file: .github/actions/compose/docker-compose.elasticsearch.yml
+          project_name: elasticsearch
+        env:
+          ELASTIC_VERSION: ${{ steps.pom-info.outputs.x_elasticsearch_test_version }}
+          ELASTIC_JVM_MEMORY: 1
+          ELASTIC_HTTP_PORT: ${{ steps.pom-info.outputs.x_new_elasticsearch_port }}
+
+      - name: Install
+        uses: ./.github/actions/run-maven
+        with:
+          parameters: install -Dskip.fe.build -Dskip.docker -DskipTests -pl optimize -am
+
+      - name: Verify migration tests
+        uses: ./.github/actions/run-maven
+        with:
+          parameters: verify -Dskip.docker -pl optimize/upgrade
+        env:
+          CAMUNDA_OPTIMIZE_DATABASE: elasticsearch
+
+      - name: Upload Test Results
+        if: always()
+        uses: actions/upload-artifact@0b2256b8c012f0828dc542b3febcab082c67f72b # v4
+        with:
+          name: migration-junit
+          path: |
+            **/failsafe-reports/**/*.xml
+            **/*.log
+            **/*.json
+          retention-days: 7
+          if-no-files-found: ignore
+
+      - name: Docker log dump
+        uses: ./.github/actions/docker-logs
+        if: always()
+        with:
+          archive_name: migration-docker
+
   # Only deploy artifacts on push event, which in this case can only be triggered by main and stable
   deploy-artifacts:
     if: github.event_name == 'push' && (github.ref_name == 'main' || startsWith(github.ref_name, 'stable/'))
     name: Deploy Artifacts
-    needs: [integration-tests]
+    needs: [ integration-tests, migration ]
     uses: ./.github/workflows/optimize-deploy-artifacts.yml
     secrets: inherit
 
   # Rerun failed jobs running on self-hosted runners in case of network issues or node preemption
   rerun-failed-jobs:
     needs:
```

Comment:
what do you think about adding these to the Unified CI? They run quickly and are stable, do they add value?

Reply:
Yeah that sounds reasonable. Guidelines say there must have been 10 green runs, but I can retrigger this one. I would then also remove them from optimize-ci so theyre not run twice on PRs right?
Wdyt about when to run? on upgrade changes obviously but there could also be backend changes that could influence the upgrade, eg if an index file is changed in a way that doesnt match the defined upgrade plan and we happen to have tests touching this, we would catch it if we run upgrade IT on general backend changes. I"m thinking of scenarios like having a new introduced field and then renaming it without adjusting the migration. Do you think its worth running them on all backend changes even though in most cases running them on all backend changes would be unnecessary?

Reply:
Added this as a separate PR, after you re-review I'll ping the CI DRI as well.

Reply:
Triggered on backend changes only seems reasonable to me. It could be a changed index file, or even anything implementing the `OptimizeDTO` interface that would break things if no migration has been written

I would be ok with this in either the unified status checks, or on the Optimize CI. It meets the status check criteria and would give us some additional confidence if there is a backend change that erroneously amended the Index DTOs, for example.

That being said, changes outside of Optimize are unlikely to break Optimize in such a way that a test failure here would be helpful. So I am also ok keeping this as part of the Optimize CI if there is resistance when you get a CI review

Reply:
Ah, I figured we were also thinking along the lines of having this in the unified to add more checks to renovate. I'll ping for a Ci review on this anyway.